tests/testthat/test-unnest.R

```diff
@@ -0,0 +1,355 @@
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars(c(y, z), values_to = "{col}_val"),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(
+        c(y, z),
+        values_to = "{col}_val"
+      ) |>
+      as.data.frame()
+  )
+})
+
+test_that("indices_to template works with multiple columns", {
+  df <- polars::pl$DataFrame(
+    x = c(1L, 2L),
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  expect_equal(
+    df |>
+      unnest_longer_polars(
+        c(y, z),
+        indices_to = "{col}_idx"
+      ),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(
+        c(y, z),
+        indices_to = "{col}_idx"
+      ) |>
+      as.data.frame()
+  )
+})
+
+test_that("both values_to and indices_to templates work together", {
+  df <- polars::pl$DataFrame(
+    x = c(1L, 2L),
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  expect_equal(
+    df |>
+      unnest_longer_polars(
+        c(y, z),
+        values_to = "{col}_val",
+        indices_to = "{col}_idx"
+      ),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(
+        c(y, z),
+        values_to = "{col}_val",
+        indices_to = "{col}_idx"
+      ) |>
+      as.data.frame()
+  )
+})
+
+test_that("no errors when col isn't list or array and return unchanged", {
+  df <- polars::pl$DataFrame(
+    id = 1:2,
+    values = list(c(1L, 2L), c(3L, 4L))
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars(id),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(id) |>
+      as.data.frame()
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars(c(id, values)),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(c(id, values)) |>
+      as.data.frame()
+  )
+})
+
+test_that("errors on non-polars data", {
+  df <- data.frame(id = 1:2, values = I(list(1:2, 3:4)))
+
+  expect_snapshot(
+    df |> unnest_longer_polars(values),
+    error = TRUE
+  )
+})
+
+test_that("errors on non-existent column", {
+  df <- polars::pl$DataFrame(
+    id = 1:2,
+    values = list(c(1, 2), c(3, 4))
+  )
+
+  expect_snapshot(
+    df |> unnest_longer_polars(nonexistent),
+    error = TRUE
+  )
+})
+
+test_that("errors when column names duplicate", {
+  df <- polars::pl$DataFrame(
+    x = c(1L, 2L),
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  expect_snapshot(
+    df |>
+      unnest_longer_polars(y, indices_to = "y"),
+    error = TRUE
+  )
+
+  expect_snapshot(
+    df |>
+      unnest_longer_polars(y, indices_to = "z"),
+    error = TRUE
+  )
+
+  expect_snapshot(
+    df |>
+      unnest_longer_polars(y, values_to = "z"),
+    error = TRUE
+  )
+
+  expect_snapshot(
+    df |>
+      unnest_longer_polars(y, indices_to = "a", values_to = "a"),
+    error = TRUE
+  )
+
+  expect_snapshot(
+    df |>
+      unnest_longer_polars(c(y, z), values_to = "{col}", indices_to = "{col}"),
+    error = TRUE
+  )
```

Comment:
I find the error messages for those cases quite unclear, e.g. 

```
Code
      unnest_longer_polars(df, y, indices_to = "z")
Condition
      Error in `unnest_longer_polars()`:
      ! Can't duplicate names between the affected columns and the original data.
      x These names are duplicated:
      i "z", from "y".
```

I think it's enough to have something like this for all tests in this block:

```
Code
      unnest_longer_polars(df, y, indices_to = "z")
Condition
      Error in `unnest_longer_polars()`:
      ! Column names in the output must be unique.
      x These names are duplicated: "z".
```

(Feel free to suggest another message)